connection_test.go

```diff
@@ -499,6 +506,38 @@ func TestWriteErrorAfterTimeout(t *testing.T) {
 	goroutines.VerifyNoLeaks(t, nil)
 }
 
+func TestWriteAfterConnectionError(t *testing.T) {
+	ctx, cancel := NewContext(time.Second)
+	defer cancel()
+
+	// Closing network connections can lead to warnings in many places.
+	opts := testutils.NewOpts().DisableLogVerification()
+	WithVerifiedServer(t, opts, func(ch *Channel, hostPort string) {
+		testutils.RegisterEcho(ch, nil)
+
+		call, err := ch.BeginCall(ctx, hostPort, ch.ServiceName(), "echo", nil)
+		require.NoError(t, err, "Call failed")
+
+		w, err := call.Arg2Writer()
+		require.NoError(t, err, "Arg2Writer failed")
+		require.NoError(t, writeFlushStr(w, "initial"), "write initial failed")
+
+		// Now close the underlying network connection, writes should fail.
+		_, conn := OutboundConnection(call)
+		conn.Close()
+
+		// Writes should start failing pretty soon.
+		var writeErr error
+		for i := 0; i < 100; i++ {
+			if writeErr = writeFlushStr(w, "f"); writeErr != nil {
+				break
+			}
+			time.Sleep(time.Millisecond)
+		}
+		assert.Error(t, writeErr, "Writes should fail after a connection is closed")
```

Comment:
Updated to check that tchannel error code is `ErrCodeNetwork`. I don't want to get too specific on exact error message, since you could get different errors like `connection reset by peer`, `use of closed network connection` etc.